### ISSUE TITLE
Could not launch process: stub exited while waiting for connection

### ISSUE THREAD

**Issue by Kosteg** [derekparker/delve#1215]
dlv version: 1.0.0
go 1.10.2 darwin/amd64
Mac OS 10.13.4, intel x64

Trying to enable debugging in vs code. Always get "could not launch process: stub exited while waiting for connection: exit status 0". When running the same command from command line – IT WORKS. But not from VS Code.

I tried every solution in previous issues:
- Installed delve using brew, manually with creating certificate, built from source code – no changes.
- Removed xcode command line tools and installed 9.2, 9.1, latest beta – no changes.

Here is my launch.json:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "remotePath": "",
            "port": 2345,
            "host": "127.0.0.1",
            "program": "${file}",
            "env": {},
            "args": [],
            "showLog": true,
            "trace": "verbose"
        }
    ]
}
```

The project/go folders:
- $GOPATH
-- bin (dlv inside)
-- src
---- github.com
---- golang.org
---- hello
------ hello.go

Here is verbose log:
```
Verbose logs are written to:
/var/folders/gq/94xhnts51vng5vs5rtd1qp1h0000gn/T/vscode-go-debug.txt
03:53:43, 2018-5-21
InitializeRequest
InitializeResponse
Using GOPATH: /Users/kosteg/go
Running: /Users/kosteg/go/bin/dlv debug hello --headless=true --listen=127.0.0.1:2345 --log=true --
2018/05/21 02:53:43 server.go:73: Using API v1
2018/05/21 02:53:43 debugger.go:101: launching process with args: [/Users/kosteg/go/src/hello/debug]
could not launch process: stub exited while waiting for connection: exit status 0
Process exiting with code: 1
Delve is closed
DisconnectRequest
```

**Comment by dotw** [comment]
thanks man!!! I have the same issue and your fix saved me.

**Comment by Kosteg** [comment]
Thanks a lot! It works now

**Comment by aarzilli** [comment]
Not merged yet.

**Comment by rubyDoomsday** [comment]
FYI: I was getting the reported error message in this thread:
```
could not launch process: stub exited while waiting for connection: exit status 0
```
whether I was starting a debug session from VSCode, GoLand, Vim or command line on a fresh install with latest GO, Delve and IDEs installed. I followed all the solutions available in, but ultimately the issue turned out to be a bad hosts file. Make sure your `/etc/hosts` is in order:
```
255.255.255.255 broadcasthost
127.0.0.1       localhost
::1             localhost
```

**Comment by alankritjoshi** [comment]
**_For anyone on M1/M2 Mac_**:

I got this same error and spent an entire day trying to figure this out.

It probably has something to do with you incorrectly installing a tool/lib **using the non-arm64 version**.  This can be `nvim`, `vs code` or an underlying dependency that is trying to create the process for `dlv`.
 
And in my case, it was due to incorrect [Brew Installation](https://docs.brew.sh/Installation). You will know if it's correct if your `brew` is somewhere in `/opt/homebrew` and NOT in `/usr/local`. Verify with `which brew`.

I [uninstalled](https://docs.brew.sh/FAQ#how-do-i-uninstall-homebrew) `brew` and pretty much everything I thought I installed with it (go, nvim, dap, delve, etc.) and also manually deleted the contents within `/usr/local` one by one (as `brew` asked).

After installing the correct version of `brew` and installing relevant stuff (go, nvim, etc.), I was able to get it working without this error.
